### PR TITLE
CS-44174 - added logger for errors for multiple files errors, version bump

### PR DESCRIPTION
**Pull request by cs-raj**
version bump:
@contentstack/cli-migration: 1.5.1 -> 1.5.2
@contentstack/cli-cm-branches: 1.0.23 -> 1.0.24

### CODE REVIEW

File: packages/contentstack-migration/src/utils/migration-logger.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+const winston = require('winston');
+const path = require('path');
+module.exports = class MigrationLogger {
```
Comment (aman19K) [inline packages/contentstack-migration/src/utils/migration-logger.js:3]: @cs-raj spacing missing

File: packages/contentstack-migration/src/utils/migration-logger.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+          level: 'error',
+          filename: path.join(this.filePath, 'error.logs'),
+          format: winston.format.combine(winston.format.timestamp(), winston.format.json()),
+        }),
+        new winston.transports.File({
+          level: 'info',
+          filename: path.join(this.filePath, 'info.logs'),
+          format: winston.format.combine(winston.format.timestamp(), winston.format.json()),
+        }),
+      ],
```
Comment (aman19K) [inline packages/contentstack-migration/src/utils/migration-logger.js:19]: @cs-raj , if info not required then please remove it

File: packages/contentstack-migration/src/utils/error-helper.js [hunk 3]
```diff
@@ -19,41 +21,58 @@ const getLineWithContext = (lines, lineNumber, context) => {
...
 
     messages.push(`${fileErrorsMessage}${errorMessages}`);
+    logger.log('error', errorLogs);
   }
   if (isEmpty(messages) && errors !== undefined && isEmpty(errorsByFile)) {
     console.error('Migration error---', errors);
+    logger.log('error', errors);
   } else {
+    logger.log('error', { error: messages.join('\n') });
     console.log(messages.join('\n'));
```
Comment (aman19K) [inline packages/contentstack-migration/src/utils/error-helper.js:76]: @cs-raj , Can we get rid of console.log and console.error now that we have logger.log?

Comment (cs-raj) [reply packages/contentstack-migration/src/utils/error-helper.js]: Removed

File: packages/contentstack-migration/src/utils/migration-logger.js (hunk 2 above)
Comment (cs-raj) [reply packages/contentstack-migration/src/utils/migration-logger.js]: Removed it and kept error logger only

File: packages/contentstack-migration/src/utils/migration-logger.js (hunk 1 above)
Comment (cs-raj) [reply packages/contentstack-migration/src/utils/migration-logger.js:3]: added

**Review by aman19K — DISMISSED** [review]

File: packages/contentstack-migration/src/commands/cm/stacks/migration.js [hunk 4]
```diff
@@ -134,6 +133,10 @@ class MigrationCommand extends Command {
     } else {
       await this.execSingleFile(filePath, mapInstance);
     }
+    const errLogPath = `${process.cwd()}/migration-logs`;
+    if (fs.existsSync(errLogPath) && !isEmpty(`${errLogPath}/error.logs`)) {
```
Comment (antonyagustine) [inline packages/contentstack-migration/src/commands/cm/stacks/migration.js:137]: `!isEmpty(`${errLogPath}/error.logs`)` - This condition is never going to be false. What are we checking here?

File: packages/contentstack-migration/src/utils/error-handler.js [hunk 5]
```diff
@@ -1,21 +1,19 @@
...
+  //   logger.log('error', { error: `Error occurred while ${errorMessageHandler[method]} ${type}: ${data}.` });
+  // }
+  // if (err.errorMessage) {
+  //   logger.log('error', { errorAPI: err.errorMessage });
+  // }
+  // if (err instanceof Error && err && err.message && err.stack) {
+  //   logger.log('error', { error: err.message });
+  // } else {
+  //   logger.log('error', { error: err });
+  // }
```
Comment (antonyagustine) [inline packages/contentstack-migration/src/utils/error-handler.js:17]: Is this code no longer required? If so we can remove the code

File: packages/contentstack-migration/src/commands/cm/stacks/migration.js (hunk 4 above)
Comment (cs-raj) [reply packages/contentstack-migration/src/commands/cm/stacks/migration.js:137]: We are checking whether the logs created have any of the error logs

Comment (cs-raj) [reply packages/contentstack-migration/src/commands/cm/stacks/migration.js]: Removed

File: packages/contentstack-migration/src/utils/error-handler.js (hunk 5 above)
Comment (cs-raj) [reply packages/contentstack-migration/src/utils/error-handler.js]: Deleted the file

**Review by aman19K — APPROVED** [review]

**Review by antonyagustine — APPROVED** [review]

**Pull request merged**